src/tests/graphql_streaming/test_api.py

```diff
@@ -306,6 +306,15 @@ def test_add_row_to_entity_add(self):
         builder._add_row_to_entity(row, entity)
         self.assertEqual(entity, expected_result)
 
+    def test_add_row_to_entity_keyerror_on_relation_from_row(self):
+        builder = self.get_instance()
+        builder.evaluation_order = ['a']
+        entity = {'some': 'entity'}
+        row = {}
+
+        with self.assertRaises(KeyError):
+            builder._add_row_to_entity(row, entity)
+
     def test_add_row_to_entity_empty_object(self):
```

Comment:
I propose to add a test specifically for _relation_from_row as it can explain much of the required behaviour

Reply:
Done :)